internal/neofs/tree.go

```diff
@@ -274,6 +276,110 @@ func (c *TreeClient) DeleteBucketCORS(ctx context.Context, cnrID *cid.ID) (*oid.
 	return nil, nil
 }
 
+func (c *TreeClient) GetObjectTagging(ctx context.Context, p *data.ObjectTaggingInfo) (map[string]string, error) {
+	_, tagNode, err := c.getObjectTaggingNode(ctx, p)
+	if err != nil {
+		return nil, err
+	}
+
+	if tagNode == nil {
+		return nil, nil
+	}
+
+	delete(tagNode.Meta, isTagKV)
+
+	return tagNode.Meta, nil
+}
+
+func (c *TreeClient) PutObjectTagging(ctx context.Context, p *data.ObjectTaggingInfo, tagSet map[string]string) error {
+	node, tagNode, err := c.getObjectTaggingNode(ctx, p)
+	if err != nil {
+		return err
+	}
+
+	if node == nil {
+		return layer.ErrNodeNotFound
+	}
+
+	tagSet[isTagKV] = "true"
+
+	if tagNode == nil {
+		_, err = c.addNode(ctx, p.CnrID, versionTree, node.ID, tagSet)
+		return err
+	}
+
+	return c.moveNode(ctx, p.CnrID, versionTree, tagNode.ID, node.ID, tagSet)
+}
+
+func (c *TreeClient) DeleteObjectTagging(ctx context.Context, p *data.ObjectTaggingInfo) error {
+	_, tagNode, err := c.getObjectTaggingNode(ctx, p)
+	if err != nil {
+		return err
+	}
+
+	if tagNode == nil {
+		return nil
+	}
+
+	return c.removeNode(ctx, p.CnrID, versionTree, tagNode.ID)
+}
+
+func (c *TreeClient) getObjectTaggingNode(ctx context.Context, p *data.ObjectTaggingInfo) (*data.NodeVersion, *TreeNode, error) {
+	var (
+		err     error
+		version *data.NodeVersion
+	)
+
+	if p.VersionID == "null" {
+		if version, err = c.GetUnversioned(ctx, p.CnrID, p.ObjName); err != nil {
+			return nil, nil, err
+		}
+	} else if len(p.VersionID) == 0 {
+		if version, err = c.GetLatestVersion(ctx, p.CnrID, p.ObjName); err != nil {
+			return nil, nil, err
+		}
+	} else {
+		versions, err := c.GetVersions(ctx, p.CnrID, p.ObjName)
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, v := range versions {
+			if v.OID.String() == p.VersionID {
+				version = v
+				break
+			}
+		}
+	}
```

Comment:
I think it's better to place getting required version in `layer` package. it will be consistent with getting objects and will simplify this function.